### PR TITLE
Add `have_cors_rule?` and `cors_rules_count` to type `s3_bucket`

### PR DESCRIPTION
**Pull request by kirikiriyamama**

### CODE REVIEW

File: lib/awspec/type/s3_bucket.rb [hunk 1]
```diff
@@ -34,5 +34,26 @@ def acl_grants_count
       @acl = find_bucket_acl(@id)
       @acl.grants.count
     end
+
+    def has_cors_rule?(allowed_headers: [], allowed_methods:, allowed_origins:, expose_headers: [], max_age_seconds: nil)
```
Comment (kirikiriyamama) [inline lib/awspec/type/s3_bucket.rb]: warning `Line is too long` occurs, do you have a workaround?

Comment (k1LoW) [inline lib/awspec/type/s3_bucket.rb]: mmm... 

you can select

```ruby
    def has_cors_rule?(allowed_headers: [],
                       allowed_methods:,
                       allowed_origins:,
                       expose_headers: [],
                       max_age_seconds: nil)
```
or
```ruby
    def has_cors_rule?(allowed_headers: [], allowed_methods:, allowed_origins:, expose_headers: [], max_age_seconds: nil) # rubocop:disable Metrics/LineLength
``` 
or 
```ruby
    # rubocop:disable Metrics/LineLength
    def has_cors_rule?(allowed_headers: [], allowed_methods:, allowed_origins:, expose_headers: [], max_age_seconds: nil)
      cors_rules.any? do |rule|
        (rule.allowed_headers - allowed_headers).empty? &&
          (rule.allowed_methods - allowed_methods).empty? &&
          (rule.allowed_origins - allowed_origins).empty? &&
          (rule.expose_headers - expose_headers).empty? &&
          rule.max_age_seconds == max_age_seconds
      end
    end
    # rubocop:enable Metrics/LineLength
```

Comment (kirikiriyamama) [inline lib/awspec/type/s3_bucket.rb]: Thank you for your advice :cat: I chose the first idea.

**Comment by k1LoW** [comment]
LGTM :+1: powerfull s3_bucket type :muscle: !!!

**Comment by k1LoW** [comment]
Released as v0.24.0.

**Pull request merged**